parquet-column/src/main/java/org/apache/parquet/schema/PrimitiveType.java

```diff
@@ -611,7 +611,7 @@ protected boolean typeEquals(Type other) {
    */
   @Override
   protected boolean equals(Type other) {
-    if (!other.isPrimitive()) {
+    if (other == null || !other.isPrimitive()) {
```

Comment:
How come the other is null? Adding a null check sounds reasonable for me, though I wonder in what situation does one call equals with a null parameter.

Reply:
It's part of the method contract: https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#equals(java.lang.Object)

This basically eliminates the need to add two checks when you want to check equality and the other object may be null. There are lots of cases where equals is used and the code doesn't need to handle null checks in addition.